### PR TITLE
std.string.lineSplitter_unittest_example

### PR DESCRIPTION
**Pull request by burner**
* added unittest doc examples for indexOf lastIndexOf

### CODE REVIEW

File: std/string.d [hunk 1]
```diff
@@ -2638,6 +2638,18 @@ auto lineSplitter(KeepTerminator keepTerm = KeepTerminator.no, Range)(Range r)
...
 }
 
+///
+@safe pure unittest
+{
+    string s = "Hello\nmy\rname\nis";
+    string[] split = splitLines(s);
+
+    int idx = 0;
+    foreach(it; lineSplitter(s)) {
```
Comment (JackStouffer) [inline std/string.d]: Why not `foreach(idx, it; s.lineSplitter.enumerate)` and get rid of the separate `idx`?

Comment (burner) [inline std/string.d]: that will not compile. lineSplitter is not an array and is not implemented with opApply(int delegate(ref size_t, ....

Comment (JackStouffer) [inline std/string.d]: It compiles just fine and passes the tests, as `enumerate` takes any input range.

Comment (JakobOvrum) [inline std/string.d]: `assert(splitLines(s).equal(lineSplitter(s)))` should work too, which I think illustrates how this example isn't very illustrating. The difference between the two is that `lineSplitter` is lazy and does not allocate; if this example is to use `splitLines`, I think this difference has to be shown somehow, or at least pointed out in a comment.

File: std/string.d [hunk 2]
```diff
@@ -2638,6 +2638,17 @@ auto lineSplitter(KeepTerminator keepTerm = KeepTerminator.no, Range)(Range r)
     return LineSplitter!(keepTerm, Range)(r);
 }
 
+///
+@safe pure unittest
+{
+    import std.algorithm.comparison : equal;
+
+    string s = "Hello\nmy\rname\nis";
+    string[] split = splitLines(s);
```
Comment (JackStouffer) [inline std/string.d]: This variable is now unused.

File: std/string.d [hunk 3]
```diff
@@ -2638,6 +2638,15 @@ auto lineSplitter(KeepTerminator keepTerm = KeepTerminator.no, Range)(Range r)
     return LineSplitter!(keepTerm, Range)(r);
 }
 
+///
+@safe pure unittest
+{
+    import std.algorithm.comparison : equal;
+
+    string s = "Hello\nmy\rname\nis";
+    assert(splitLines(s).equal(lineSplitter(s)));
```
Comment (DmitryOlshansky) [inline std/string.d]: This is fine but maybe a comment on lazyiness of  lineSplitter vs splitLines? e.g. splitLines being an array can be sorted directly.

Comment (JakobOvrum) [inline std/string.d]: Yeah. Like I already said, I don't think this is a useful example at all. Going by this example alone, it just begs the question "so what's the difference?".

**Comment by burner** [comment]
thanks

**Pull request merged**